lib/backends/sdc/containers.js

```diff
@@ -726,97 +765,68 @@ function namedNetwork(opts, container, payload, callback) {
-            } else if (exactName && !idMatch) {
-                // wins if we don't have an id match.
-                acc = nw;
-            } else if (acc && !idMatch && !nameMatch) {
-                // logically !exactId, !exactName. If we have not already found
-                // a better match, this is the *second* partial match, and we
-                // need to error unless we find a better one.
-                acc = new errors.AmbiguousDockerNetworkIdPrefixError(query, c);
+        // results.operations is ordered per the funcs array provided
+        // to vasync.parallel (see vasync docs). We can tolerate partial
+        // errors as long as they are lower in the preference chain.
+        // IOW, we callback with the err/result of the most-preferred
+        // search, and log any additional errors.
+        var bestMatch = results.operations.reduce(function (acc, op) {
+            if (acc.err || acc.result) {
+                return acc;
+            }
+            if (op.err) {
+                acc.err = op.err;
+                return acc;
+            }
+            // all match funcs are listNetworks, contract is to return an
+            // err (handled above) or array.
+            switch (op.result.length) {
+            case 0:
+                break;
+            case 1:
+                acc.result = op.result[0];
+                break;
+            default:
+                acc.err = new errors.AmbiguousDockerNetworkIdPrefixError(query);
+                break;
             }
-
             return acc;
-        });
+        }, { err: null, result: null });
 
+        if (bestMatch.err) {
+            // found an error before a result.
+            log.error({ err: err, query: query },
+                format('Networks: Error finding network to match %s', query));
+            callback(err);
+            return;
+        }
 
-        if (network.restCode) {
-            // actually an error.
-            log.error({ err: network }, 'Networks: ambiguous result for %s',
-                query);
-            return callback(network);
+        if (!bestMatch.err && !bestMatch.result) {
+            log.info({ query: query, user: opts.account.uuid },
+                format('Networks: no results for query %s', query));
+            callback(new errors.NetworkNotFoundError(query));
+            return;
         }
 
-        log.debug({ network: network }, 'Networks: chose %s/%s',
-            network.name, network.uuid);
+        if (!bestMatch.err && err) {
+            // found result before an error, but did have errs.
+            log.warn({ err: err },
+                'Networks: non-critical error searching NAPI');
+        }
 
+        log.debug({ network: bestMatch.result }, 'Networks: chose %s/%s',
+            bestMatch.result.name, bestMatch.result.uuid);
 
-        payload.networks = [ { uuid: network.uuid, primary: true } ];
+        payload.networks = [ { uuid: bestMatch.result.uuid, primary: true } ];
```

Comment:
Ah, got it: not it's not possible because of https://github.com/joyent/sdc-docker/pull/87/commits/bf5c44f5fbcac91f6cb590f8aa1d3d19329fbdbb#diff-d36e9a28d5d9b3702ee9fa48d5326b0dR1959 :)